src/nnet3/nnet-component-itf.cc

```diff
@@ -251,7 +251,6 @@ void NonlinearComponent::ZeroStats() {
 
 std::string NonlinearComponent::Info() const {
   std::stringstream stream;
-  KALDI_ASSERT(InputDim() == OutputDim());  // always the case
   stream << Type() << ", dim=" << InputDim();
```

Comment:
You should extend the Info(), so that it's clear from the info string whether you have this extra feature.
Probably if the feature is included, it should say input-dim=xxx, output-dim=xxx, include-log-stddev=true